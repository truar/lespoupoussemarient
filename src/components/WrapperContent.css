
.WrapperContent {
    justify-content: space-evenly;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.WrapperContent p, .WrapperContent ul {
    font-size: 1rem;
    margin: 0px;
    padding: 0px;
}

.WrapperContent h2 {
    text-align: center;
}

.WrapperContent h3 {
    font-size: 1.5rem;
}

@media (min-width: 768px) {
    .WrapperContent p, .WrapperContent ul {
        font-size: 1.5rem;
    }
    .WrapperContent h3 {
        font-size: 2rem;
    }
}

@media (min-width: 992px) {
    .WrapperContent p {
        font-size: 1.75rem;
    }
}
