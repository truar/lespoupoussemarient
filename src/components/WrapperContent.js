import React from 'react';
import './WrapperContent.css';

function WrapperContent(props) {
    const {children, id, title} = props;

    return (
        <div id={id} className="WrapperContent">
            {title &&
                <h2>
                    {title}
                </h2>
            }
            {children}
        </div>
    )
}

export default WrapperContent;