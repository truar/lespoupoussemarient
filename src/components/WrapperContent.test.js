import React from 'react';
import WrapperContent from './WrapperContent';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const component = shallow(<WrapperContent />);
  expect(component).toMatchSnapshot();
});

it('renders with a title', () => {
  const component = shallow(<WrapperContent title="Title" />);
  expect(component).toMatchSnapshot();
});