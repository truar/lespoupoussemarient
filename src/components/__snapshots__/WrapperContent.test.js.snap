// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with a title 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WrapperContent
    title="Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Title
        </h2>,
        undefined,
      ],
      "className": "WrapperContent",
      "id": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
        },
        "ref": null,
        "rendered": "Title",
        "type": "h2",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Title
          </h2>,
          undefined,
        ],
        "className": "WrapperContent",
        "id": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title",
          },
          "ref": null,
          "rendered": "Title",
          "type": "h2",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WrapperContent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
      ],
      "className": "WrapperContent",
      "id": undefined,
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
        ],
        "className": "WrapperContent",
        "id": undefined,
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
